modules/custom-status/custom-status.php

```diff
@@ -1314,5 +1347,17 @@ public function fix_post_row_actions( $actions, $post ) {
 		$actions['view'] = '<a href="' . esc_url( $preview_link ) . '" title="' . esc_attr( sprintf( __( 'Preview &#8220;%s&#8221;' ), $post->post_title ) ) . '" rel="permalink">' . __( 'Preview' ) . '</a>';
 		return $actions;
 	}
+
+	public function add_custom_status_ui_properties( $custom_status ) {
+		// Add 'required_user_logins' data with display names for status restrictions UI
+		$required_user_ids = $custom_status->required_user_ids ?? [];
+
+		$custom_status->required_user_logins = array_filter(array_map( function ( $user_id ) {
```

Comment:
If you port over the `get_term` and `get_terms` filters from #41 you can move this code there. It would be targeted at just the custom status taxonomy, and would ensure that the data was available. The only change would be that the data would be under the meta field, rather than being directly accessible.

Happy to also do that in a follow up PR from my end tbh.

Reply:
Scratch that, don't do this approach. I assumed that I'd be able to set different priorities on editorial metadata and custom status so that I could have different get_term and get_terms calls. But that has failed and there is a way to work around this, which is to have the two modules share code. That defeats the purpose of trying to keep them isolated. So its better to just have the get and get_by calls within the individual modules be the source of truth for any retrievals.